pkg/virt-controller/watch/annotations.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package watch
+
+import (
+	"errors"
+	"fmt"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	k8sv1 "k8s.io/api/core/v1"
+
+	virtv1 "kubevirt.io/api/core/v1"
+)
+
+func (c *VMIController) copyMultusAnnotationsFromPodToVmi(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
+	annotationsKeyList := []string{networkv1.NetworkStatusAnnot}
+	if annotationsCopiedToVmi(vmi, pod, annotationsKeyList) {
+		return nil
+	}
+
+	rateLimitedExecutor := c.sriovVmiCopyAnnotExecPool.LoadOrStore(vmi.UID)
+	if rateLimitedExecutor.IsTimeout() {
+		return nil
+	}
+	rateLimitedExecutor.Exec(func() error {
+		var err error
+		for _, annotationKey := range annotationsKeyList {
+			err = copyPodAnnotationToVmi(vmi, pod, annotationKey)
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+
+	if !annotationsCopiedToVmi(vmi, pod, annotationsKeyList) {
+		return fmt.Errorf("annotations %v not copied to vmi", annotationsKeyList)
+	}
+
```

Comment:
As we've talked offline, another possible solution is to add a returned value RateLimitedExecutor.Exec(), so the caller could distinguish between two scenarios:
1. The `command` function was executed and completed successfully.
2. The `command` function was not executed, because the backoff was not ready.

With this added functionality, the second call to `annotationsCopiedToVmi()` could be dropped.

Reply:
Yes, I agree but I think in this PR we can make due with the current functionality of the rateLimitedExecutor.